server/world/mcdb/provider.go

```diff
@@ -141,6 +144,85 @@ func (p *Provider) SaveSettings(s *world.Settings) {
 	p.saveDifficulty(s.Difficulty)
 }
 
+// LoadPlayerSpawnPosition loads the players spawn position stored in the level.dat from their UUID.
+func (p *Provider) LoadPlayerSpawnPosition(uuid uuid.UUID) (pos mgl64.Vec3, exists bool, err error) {
+	data, err := p.db.Get(append([]byte("player_"), uuid[:]...), nil)
+	if err != nil {
+		return mgl64.Vec3{}, false, err
+	}
+	var playerData map[string]string
+
+	if err := nbt.UnmarshalEncoding(data, &playerData, nbt.LittleEndian); err != nil {
+		return mgl64.Vec3{}, false, err
+	}
+	if playerData["MsaID"] != uuid.String() && playerData["ServerId"] == "" {
+		return mgl64.Vec3{}, false, fmt.Errorf("LoadPlayerSpawn: Invalid data for player %s", uuid.String())
```

Comment:
Would make this lowercase to be consistent with Go convention. (e.g. `load player spawn: invalid data for player: %v`)